src/NexusMods.Library/InstallLoadoutItemJobWorker.cs

```diff
@@ -23,32 +21,38 @@ public InstallLoadoutItemJobWorker(IServiceProvider serviceProvider)
     protected override async Task<JobResult> ExecuteAsync(InstallLoadoutItemJob job, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
-        if (!job.Installer.HasValue)
-        {
-            ILibraryItemInstaller? foundInstaller = null;
-            var installers = job.Loadout.InstallationInstance.GetGame().LibraryItemInstallers;
-            foreach (var installer in installers)
-            {
-                var isSupported = await installer.IsSupportedAsync(job.LibraryItem, cancellationToken);
-                if (!isSupported) continue;
 
-                foundInstaller = installer;
-                break;
-            }
+        var installers = job.Loadout.InstallationInstance.GetGame().LibraryItemInstallers;
+        var result = await ExecuteInstallersAsync(job, installers, cancellationToken);
 
+        if (result is null)
+        {
             // TODO: default to advanced installer
-            if (foundInstaller is null)
-            {
-                return JobResult.CreateFailed($"Found no installer that supports `{job.LibraryItem.Name}` (`{job.LibraryItem.Id}`)");
-            }
-
-            job.Installer = Optional<ILibraryItemInstaller>.Create(foundInstaller);
+            return JobResult.CreateFailed($"Found no installer that supports `{job.LibraryItem.Name}` (`{job.LibraryItem.Id}`)");
         }
 
-        var result = await job.Installer.Value.ExecuteAsync(job.LibraryItem, job.Transaction, job.Loadout, cancellationToken);
         var transactionResult = await job.Transaction.Commit();
-
         var jobResults = result.Select(x => transactionResult.Remap(x)).ToArray();
         return JobResult.CreateCompleted(jobResults);
     }
+
+    private static async ValueTask<LoadoutItem.New[]?> ExecuteInstallersAsync(
+        InstallLoadoutItemJob job,
+        ILibraryItemInstaller[] installers,
+        CancellationToken cancellationToken)
+    {
+        foreach (var installer in installers)
+        {
+            var isSupported = installer.IsSupportedLibraryItem(job.LibraryItem);
+            if (!isSupported) continue;
+
+            // TODO: we need to retract all added entities in the transaction if the installer doesn't support the item
```

Comment:
Maybe we just restart the transaction in that case?

Reply:
What do you mean with "restart"?